superset/security/api.py

```diff
@@ -160,3 +163,33 @@ def guest_token(self) -> Response:
             return self.response_400(message=error.message)
         except ValidationError as error:
             return self.response_400(message=error.messages)
+
+    @expose("/custom_login/", methods=["GET"])
+    @event_logger.log_this
+    @safe
+    @permission_name("read")
+    def custom_login(self) -> Response:
+        token_val = request.headers.get("Authorization")
+
+        headers = {"Authorization": token_val, "accept": "application/json"}
+
+        response = requests.get(
+            f"{current_app.config.get('PLATFORM_RBAC_BASE_URL')}/v1/users/me?expand=organizations.accounts",
+            headers=headers,
+        )
+
+        if response.status_code != 200:
+            return self.response(400)
+
+        email = response.json().get("email")
+
+        if not email:
+            return self.response(400)
+
+        user = security_manager.find_user(email=email)
+
+        login_user(user, remember=False)
+
+        dashboard_id = request.args.get("dashboard_id")
+
+        return redirect(f"{request.url_root}/{dashboard_id}")
```

Comment:
As we don't want users to land on a page different than the Dashboard one, can we reject the request and avoid the redirect if the `dashboard_id` is undefined ?

Reply:
done